llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6053,9 +6053,38 @@ Value *SPIRVToLLVM::transSPIRVImageQueryLodFromInst(SPIRVInstruction *bi, BasicB
   // A sampler descriptor is encoded as {desc,convertingSamplerIdx}. Extract the actual sampler.
   Value *samplerDesc = getBuilder()->CreateExtractValue(imageInfo.samplerDesc, 0);
 
-  // Generate the operation.
+  // Generate the operation for normal image get lod.
   Value *coord = transValue(bii->getOpValue(1), bb->getParent(), bb);
-  return getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+  Value *result =
+      getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+
+  // Extract sampler descriptor for ycbcr, otherwise the desc would be null.
+  if (!m_convertingSamplers.empty()) {
+    // Refer to transSPIRVImageSampleFromInst for details.
+    Value *convertingSamplerIdx = getBuilder()->CreateExtractValue(imageInfo.samplerDesc, 1);
+    int thisConvertingSamplerIdx = 1;
+    for (const ConvertingSampler &convertingSampler : m_convertingSamplers) {
+      unsigned arraySize = convertingSampler.values.size() / ConvertingSamplerDwordCount;
+      for (unsigned idx = 0; idx != arraySize; ++idx) {
+        const unsigned samplerDescSize = 4;
+        SmallVector<Constant *, samplerDescSize> samplerInts;
+        for (unsigned component = 0; component != samplerDescSize; ++component) {
+          samplerInts.push_back(
+              getBuilder()->getInt32(convertingSampler.values[idx * ConvertingSamplerDwordCount + component]));
+        }
+        samplerDesc = ConstantVector::get(samplerInts);
+        Value *thisResult =
+            getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+        // Add to select ladder.
```

Comment:
Can you move this comment up to the for loop that constructs the series of selects?

Reply:
Moved, simplifed.